### ISSUE TITLE
Make Java installations ensurable (present/absent)

### ISSUE THREAD

**Issue by staylorx** [antoineco/aco-oracle_java#28]
I expected an 'ensure' parameter but didn't find one. Nice-to-have.

And again, thank you for the module.

**Comment by antoineco** [comment]
Indeed, I've seen this pattern a while ago, some people used to state entire modules should be ensurable, some others disagreed. In general I would agree with the idea as long as it doesn't introduce some unnecessary complexity (e.g. forcing the module to get split in "installer" and "uninstaller" classes.

Also this module includes some archive extraction for the _tag.gz_ installation type, and the opposite of an archive extraction is a directory deletion, so we have to make sure we delete the right directory and not some extra files :) Puppet's [https://docs.puppet.com/puppet/latest/reference/types/tidy.html](tidy) resource can certainly do that.

Once again I'm open to suggestions.
